hawkeye-api/src/backend.rs

```
@@ -0,0 +1,283 @@
+    watcher_id: &str,
+) -> kube::Result<Deployment> {
+    let deployments_client: Api<Deployment> =
+        Api::namespaced(k8s_client.clone(), &config::NAMESPACE);
+    deployments_client
+        .get(&templates::deployment_name(watcher_id))
+        .await
+}
+
+/// Scale a Watcher's Kubernetes deployment by altering the number of replicas. Great for turning down to 0.
+pub async fn scale_watcher_deployment(
+    k8s_client: &kube::Client,
+    watcher_id: &str,
+    replica_count: u16,
+) {
+    let patch_params = PatchParams {
+        field_manager: Some(FIELD_MGR.to_owned()),
+        ..Default::default()
+    };
+    let deployment_scale_json = json!({
+        "apiVersion": "autoscaling/v1",
+        "spec": {"replicas": replica_count},
+    });
+    let deployments_client: Api<Deployment> =
+        Api::namespaced(k8s_client.clone(), &config::NAMESPACE);
+    deployments_client
+        .patch_scale(
+            &templates::deployment_name(watcher_id),
+            &patch_params,
+            &Patch::Merge(&deployment_scale_json),
+        )
+        .await
+        .unwrap();
+}
+
+/// Update a Watcher's status to indicate it should be running or not.
+pub async fn update_watcher_deployment_target_status(
+    k8s_client: &kube::Client,
+    watcher_id: &str,
+    status: Status,
+) {
+    let patch_params = PatchParams {
+        field_manager: Some(FIELD_MGR.to_owned()),
+        ..Default::default()
+    };
+    let status_label_json = json!({
+        "apiVersion": "apps/v1",
+        "metadata": {
+            "labels": {
+                "target_status": status,
+            }
+        }
+    });
+    let deployments_client: Api<Deployment> =
+        Api::namespaced(k8s_client.clone(), &config::NAMESPACE);
+    deployments_client
+        .patch(
+            &templates::deployment_name(watcher_id),
+            &patch_params,
+            &Patch::Merge(status_label_json),
+        )
+        .await
+        .unwrap();
```

Same here, probably should return an error

---

updated to return a Result